### ISSUE TITLE
Lesson 09: The general rule is confusing

### ISSUE THREAD

**Issue by delcypher** [halide/Halide#945]
The "general rule" is quite confusing

```C++
        // The general rule is: Each Var used in an update definition
        // must appear unadorned in the same position as in the pure
        // definition in all references to the function on the left-
        // and right-hand sides. So the following definitions are
        // legal updates:
```

``f(x, x + 1) = x + 8;`` seems like a counter example, ``x +1`` doesn't seem like it should be usable where it is because in the pure definition ``y`` is at that argument position.

``f(y/2, y) = f(0, y) * 17;`` seems like a counter example. Ignoring ``y/2`` being in argument position used for ``x``, the other issue is that ``y/2`` is "adorned", it's not just a plain ``y``.

**Comment by abadams** [comment]
Yeah, I struggled with how to phrase this. If you use 'x', then all
references to the function must have 'x' unadorned in its place. But x can
also be used elsewhere, which is what those two examples demonstrate.
